File: dotnet-maui/ReservationsDotnetMaui/Models/Reservation.cs
namespace ReservationsDotnetMaui.Models;

public class Reservation
{
    public string Id { get; set; } = string.Empty;
    public string UserId { get; set; } = string.Empty;
    public string OfficeId { get; set; } = string.Empty;
    public string? DeskId { get; set; }
    public string? ConferenceRoomId { get; set; }
    public DateTime ReservedAt { get; set; }
    public DateTime ReservedUntil { get; set; }
    public bool IsActive { get; set; } = true;
}

File: flutter/flutterapp/lib/models/reservation.dart
class Reservation {
  Reservation({
    required this.id,
    required this.userId,
    required this.officeId,
    this.deskId,
    this.conferenceRoomId,
    required this.reservedAt,
    required this.reservedUntil,
    this.isActive = true,
  });

  final String id;
  final String userId;
  final String officeId;
  final String? deskId;
  final String? conferenceRoomId;
  final DateTime reservedAt;
  final DateTime reservedUntil;
  bool isActive;
}
